Lib/test/test_typing.py

```diff
@@ -2611,6 +2611,10 @@ def test_forward_equality(self):
         fr = typing.ForwardRef('int')
         self.assertEqual(fr, typing.ForwardRef('int'))
         self.assertNotEqual(List['int'], List[int])
+        self.assertNotEqual(fr, typing.ForwardRef('int', module=__name__))
```

Comment:
Hmm, I wonder if this test is run directly, will `__name__` be `__main__``? Wouldn't that make the test pass incorrectly? (Throwing this question out there because I'm not too sure myself). Maybe we should be safe and give it a distinct name?

Reply:
The test is to ensure that `ForwardRef` with `module` set is different from `ForwardRef` with `module` not set (That was previsously not the case). The actual value of `module` (or `__name__`) is not relevant. Under the hood the comparison is `None` vs `str`

Reply:
My bad, thanks for pointing that out. I mistakenly assumed it would grab a default value from the current scope to populate `module` in things like `ClassVar['int']` due to the `_type_check`/`_type_convert` function (I'm now shocked that it doesn't and only applies to `TypedDict`).

Reply:
Yes, unfortunately the current module is not captured automatically.
The current scope/module is typicallly obtained by `get_type_hints` because a class knows its module (and so all ForwardRefs in member annotations can be correctly resolved. But each forward Ref does not know its module). 
However for type aliases this does not work, and there one needs the explicit module.

For example

Json = Union[ List['Json'], Dict[str,'Json'], int, float, bool, None ]

cannot be exported to a different module unless one sets the `module` for each forward ref.